### DISCUSSION TITLE
Automate - used for filter, sort, group by, ... ??

### DISCUSSION THREAD

**Discussion by olegrand1976** [Budibase/budibase#6981] [Help]
Hello to all,

    I would like to create 2 fields in my table that would be filled by automaton when a new row is created and that I can then use these fields in the filters of my views.

I explain:

1) In a first step, I had created : 

(I only include below the fields that are useful for the explanation)

TABLE: contract
Field_1 : premium
Field_2: employee_commission
Filed_3: commission_net

TABLE: contract_type
Field_1: commission_rate

TABLE: employee
Field_1: ratio

Here are the calculations I need to perform:

contract.commission_collaborator = contract.premium * collaborator.ratio

commission_nette = contract.premium * employee.ratio * contract_type.commission_rate

If I use them in "formula" fields, I don't know how to use them afterwards (filter, sort, group by, ...).

2) Then I try "automate" but I get lost and I don't understand how to code and I don't know if I could use them after (filter, sort, group by, ...).

Could you guide me? :)

Have a nice day

**Comment by melohagan** [comment]
Hi @olegrand1976 

Formulas seem appropriate for these types of calculations. Where you finding that the formula fields were not appearing as filter options?
Are you using the internal Budibase DB or an external DB?
